### ISSUE TITLE
Lighting engine, strange behavior.

### ISSUE THREAD

**Issue by XanCraft21** [C-Chads/tinygl#28]
Hello, i would like to report a possible bug with the lighting engine.

This may be because i am forgetting a setting i need to add, but if i shine a light on a 3D model, a cube to be exact, as the model rotates the light flashes on and off depending on the rotation angle.

It seems like the light is not taking the model’s rotation angle into consideration before rendering the light effect. Is this a bug or is there a setting i am forgetting to set in my code?

If you need a video let me know. Thank you for your help.

**Comment by XanCraft21** [comment]
I have figured out the problem, i did not define any normals for my model. Although now the lighting engine no longer smoothly shades the model, but only the flat triangles themselves and not at each pixel.

I want to keep this open because i need to ask for help on how i can make the light think there are normals to get the smooth lighting. I got the proper smoothing and positing working by changing when and how i set the light position, but it’s like the light spins around the model making it appear as a specific part of the model is lit regardless of rotation.

Is there a way to make Tinygl automatically calculate the normals for each triangle of my model? Thank you for your help.
